tools/runners/run-app.js

```diff
@@ -835,8 +848,14 @@ _.extend(AppRunner.prototype, {
 
     Console.enableProgressDisplay(false);
 
-    const postStartupResult = runPostStartupCallbacks(bundleResult);
-    if (postStartupResult) return postStartupResult;
+    const postStartupResult = runPostStartupCallbacks({
+      bundleResult,
+      cancelToken: CancelToken.fromPromise(runPromise),
```

Comment:
This essentially races the post-startup callbacks against the `runPromise`, rather than running all post-startup callbacks before awaiting the result of the `runPromise`.